src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs

```diff
@@ -43,33 +46,51 @@ private static VersionInfo GetVersionInfo()
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(MatrixFactorizationPredictor).Assembly.FullName);
+                loaderAssemblyName: typeof(MatrixFactorizationModelParameters).Assembly.FullName);
         }
         private const uint VersionNoMinCount = 0x00010002;
 
         private readonly IHost _host;
-        // The number of rows.
-        private readonly int _numberOfRows;
-        // The number of columns.
-        private readonly int _numberofColumns;
-        // The rank of the factor matrices.
-        private readonly int _approximationRank;
-        // Packed _numberOfRows by _approximationRank matrix.
+        ///<summary> The number of rows.</summary>
+        public readonly int NumberOfRows;
+        ///<summary> The number of columns.</summary>
+        public readonly int NumberOfColumns;
+        ///<summary> The rank of the factor matrices.</summary>
+        public readonly int ApproximationRank;
+        // Packed NumberOfRows by ApproximationRank matrix.
         private readonly float[] _leftFactorMatrix;
-        // Packed _approximationRank by _numberofColumns matrix.
+        // Packed ApproximationRank by NumberOfColumns matrix.
         private readonly float[] _rightFactorMatrix;
 
-        public PredictionKind PredictionKind
+        /// <summary>
+        /// Returns left approximation matrix.
+        /// </summary>
+        /// <returns>Flattened into one-dimensional array with size equal to <see cref="NumberOfRows"/> * <see cref="ApproximationRank"/></returns>
+        public float[] GetLeftFactorMatrix()
         {
```

Comment:
Any test for the two public functions, GetLeftFactorMatrix and GetRightFactorMatrix?

Reply:
yes, I modified one of tests see MatrixFactorizationTests.cs

---
In reply to: [254907007](https://github.com/dotnet/machinelearning/pull/2451#discussion_r254907007) [](ancestors = 254907007)